### PR TITLE
ycmd: 2019-09-19 -> 2020-02-22

### PR DESCRIPTION
**Pull request by evanjs**
###### Motivation for this change
It seems YCMD was left behind (from YCM) with #80218.
This normally might not make a difference, but this time, they came out of sync with commits that dropped python2 support.

The commits mentioned are https://github.com/ycm-core/YouCompleteMe/pull/3566 and https://github.com/ycm-core/ycmd/pull/1382, both of which were merged within the last month.

This broke YouCompleteMe, throwing the message:
`YouCompleteMe unavailable: No module named 'future'`

###### Things done
This PR updates YCMD and disables python2 support for both YCM and YCMD.

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by rasendubi** [comment]
The diff looks good, but does last commit need to go with ycmd update? It looks like it touches quite some things beside ycmd

**Comment by evanjs** [comment]
> The diff looks good, but does last commit need to go with ycmd update? It looks like it touches quite some things beside ycmd

Assuming that `vimPlugins` is to be updated with PRs pertaining to any packages in the set, then yes. 

I know we need to update the set for plugin additions, though I’m not sure about plugin updates. 
[Vim Section Docs](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/vim.section.md)

This is assuming that the change for dropping python2 support for youcompleteme should be done in this PR, anyway.
As YCMD is a dependency of YCM, I felt this was acceptable.

File: pkgs/misc/vim-plugins/overrides.nix [hunk 1]
```diff
@@ -471,6 +471,7 @@ self: super: {
   });
 
   youcompleteme = super.youcompleteme.overrideAttrs(old: {
+    disabled = !python.isPy3k;
```
Comment (jonringer) [inline pkgs/misc/vim-plugins/overrides.nix]: pretty sure this is unnecessary since you made the change in the python-module as well

Comment (evanjs) [reply pkgs/misc/vim-plugins/overrides.nix]: I disabled python2 support in both `ycmd` and `youcompleteme`, as they're two separate modules, that link back to two separate repositories with python2 support explicitly dropped in each.

I'm fine with whatever, though, and if it makes more sense to remove the `disabled` bit here, I'd be happy to do so.

Comment (jonringer) [reply pkgs/misc/vim-plugins/overrides.nix]: I'm not sure if this does anything, the `disabled = ...` is specific to `buildPython{Application,Package}`

Comment (evanjs) [reply pkgs/misc/vim-plugins/overrides.nix]: Sounds good.  Didn't realize it was `buildPython` specific.

**Review by jonringer — APPROVED** [review]
diff LGTM
commit LGTM

```
[41 built, 25 copied (193.7 MiB), 44.9 MiB DL]
https://github.com/NixOS/nixpkgs/pull/81034
36 package built:
vimPlugins.SpaceCamp vimPlugins.ale vimPlugins.awesome-vim-colorschemes vimPlugins.coc-git vimPlugins.coc-java vimPlugins.coc-metals vimPlugins.defx-nvim vimPlugins.denite-git vimPlugins.deoplete-fish vimPlugins.deoplete-khard vimPlugins.deoplete-nvim vimPlugins.echodoc-vim vimPlugins.falcon vimPlugins.jedi-vim vimPlugins.ncm2-ultisnips vimPlugins.neosnippet-snippets vimPlugins.nvim-lsp vimPlugins.seoul256-vim vimPlugins.ultisnips vimPlugins.unicode-vim vimPlugins.vim-airline vimPlugins.vim-codefmt vimPlugins.vim-dirvish vimPlugins.vim-fugitive vimPlugins.vim-gitgutter vimPlugins.vim-go vimPlugins.vim-localvimrc vimPlugins.vim-slime vimPlugins.vim-snippets vimPlugins.vim-terraform vimPlugins.vim-test vimPlugins.vim-visual-multi vimPlugins.vimtex vimPlugins.vista-vim vimPlugins.youcompleteme ycmd
```

**Comment by jonringer** [comment]
@rasendubi, @evanjs is right, it's common practice to update all vimplugins before adding a new plugin

**Comment by evanjs** [comment]
> @rasendubi, @evanjs is right, it's common practice to update all vimplugins before adding a new plugin

Thanks for clarifying! I wasn’t certain if this also applies to updates!

**Pull request merged**